File: cash.py
from cs50 import get_float

while True:
    change = get_float("Change owed: ")
    if change >= 0:
        break

# Convert change to cents
cents = int(change * 100)

# Number of quarters
# // 整数除法取整运算
quarters = cents // 25
# % 整数除法余数运算
cents %= 25

# Number of dimes
dimes = cents // 10
cents %= 10

# Number of nickels
nickels = cents // 5
cents %= 5

# Number of pennies
pennies = cents

# Total number of coins
total = quarters + dimes + nickels + pennies
print(total)

File: cash.c
#include<stdio.h>
#include<cs50.h>

int main (void)
{
    // Prompt for users to enter positive heights
    int n;
    do
    {
        n = get_int("Change owed: ");
    }
    while (n < 1);

    // Find number of 25 cents
    int a = n / 25;

    n = n - 25 * a;

    // Find the number of 10 cents
    int b = n / 10;

    n = n - 10 * b;

    // Find the number of 5 cents
    int c = n / 5;

    n = n - 5 * c;

    // Find the number of 1 cent
    int d = n;

    printf("%i\n", a + b + c + d);

}




